includes/data-stores/class-wc-customer-data-store.php

```diff
@@ -132,7 +132,16 @@ public function read( &$customer ) {
 		}
 
 		$customer_id = $customer->get_id();
-		$customer->set_props( array_map( 'wc_flatten_meta_callback', get_user_meta( $customer_id ) ) );
+		// Load meta but exclude deprecated props.
+		$user_meta   = array_diff_key( array_map( 'wc_flatten_meta_callback', get_user_meta( $customer_id ) ), array(
```

Comment:
I think it might be a bit cleaner to do something like `array_flip( array( 'country', 'state', 'postcode', 'city', 'address', 'address_2' ) )` here to help reduce the number of lines -- but the actual logic here LGTM.